### PR TITLE
Feature/autoreset

### PR DESCRIPTION
**Pull request by rsinnet**
Fixes MisoRobotics/grillbot#169.

### CODE REVIEW

**Comment by rsinnet** [comment]
Actually, also need to run the usb reset function, so that has to be called. Should probably be called in `kinect2_bridge.cpp`.

**Comment by rsinnet** [comment]
This code in the pull request is needed to restart the `kinect2_bridge` node. To finish the implementation of auto-restarting we need to reset the USB bus somehow. @wwerst found that setting the usbfs memory size after a crash reset the kinect at least the one time he tried it. See also:
* https://github.com/OpenKinect/libfreenect2/wiki/Troubleshooting#multiple-kinects-try-increasing-usbfs-buffer-size
* https://www.ptgrey.com/tan/10685

**Comment by wwerst** [comment]
The usbfs memory size may have also been the root cause of this issue. It is now increased to 128mb by default on flippy-desktop (via grub). Time will tell whether it is likely the root cause.

**Comment by wwerst** [comment]
The kinect crashed again. I increased memory again. This caused the kinects to start working again.

File: kinect2_bridge/src/kinect2_bridge.cpp [hunk 1]
```diff
@@ -914,10 +915,20 @@ class Kinect2Bridge
         if(isSubscribedDepth)
         {
           OUT_INFO("depth processing: " FG_YELLOW "~" << (tDepth / framesIrDepth) * 1000 << "ms" NO_COLOR " (~" << framesIrDepth / tDepth << "Hz) publishing rate: " FG_YELLOW "~" << framesIrDepth / fpsTime << "Hz" NO_COLOR);
+          if (framesColor == 0)
```
Comment (zzv2) [inline kinect2_bridge/src/kinect2_bridge.cpp]: This should be framesIrDepth, right?

**Comment by zzv2** [comment]
#6 is this except the nodelet version.

**Pull request closed without merging**